src/os/layer/preset/abstractpresetservice.js

```diff
@@ -0,0 +1,213 @@
+goog.module('os.layer.preset.AbstractPresetService');
+
+
+const IPresetService = goog.requireType('os.layer.preset.IPresetService');
+
+/**
+ * Class to provide stubs and simple implementation of setDefault() and setPublished()
+ *
+ * @abstract
+ * @implements {IPresetService}
+ */
+class AbstractPresetService {
+  /**
+   * @inheritDoc
+   */
+  insert(preset) {
+    return new Promise((resolve) => {
+      resolve(null);
+    });
+  }
+
+  /**
+   * @inheritDoc
+   */
+  update(preset) {
+    return new Promise((resolve) => {
+      resolve(null);
+    });
+  }
+
+  /**
+   * @inheritDoc
+   */
+  remove(preset) {
+    return new Promise((resolve) => {
+      resolve(null);
+    });
+  }
+
+  /**
+   * @inheritDoc
+   */
+  find(search) {
+    return new Promise((resolve) => {
+      resolve(null);
+    });
+  }
+
+  /**
+   * Re-set the params and return instead of inheritDoc.
+   * The compiler is wrong; preset, opt_boolean, and the return are exactly as they should be.
+   *
+   * @param {osx.layer.Preset|string} preset
+   * @param {boolean=} opt_boolean
+   * @return {!Promise<osx.layer.Preset|null,?>}
```

Comment:
If you want to make one last effort at using `@inheritDoc` here, the only other weird thing is the `,?` part of the type annotation.  I don't really know what that's supposed to mean in the context of a solution of a promise. `?osx.layer.Preset` makes sense as "null or an osx.layer.Preset", but the comma is weird.